src/controller.rs

```diff
@@ -40,6 +40,11 @@ impl Controller {
         // Start the dummy adapter.
         let dummy_adapter = DummyAdapter::new(self.sender.clone(), self.context.clone());
         dummy_adapter.start();
+
+        {
+            let mut context = self.context.lock().unwrap();
+            context.start_tunnel().unwrap();
+        }
```

Comment:
This is at the end of the function block and owned variables are about to go out of scope, what does the nested scope grant us here?